### PR TITLE
Sagesse

### PR DESCRIPTION
**Pull request by raphell**

### CODE REVIEW

File: src/controllers/sagesseController.ts [hunk 1]
```diff
@@ -0,0 +1,234 @@
+const Request = require('request')
+const jsdom = require('jsdom')
+const { JSDOM } = jsdom
+
+
+export const getSubjectDetails = async (subjectId: string, year: string, res: any) => {
```
Comment (MartinCayuelas) [inline src/controllers/sagesseController.ts]: for res : any you should use the Response type of the module 'express'
import {Response} from 'express;

File: src/controllers/sagesseController.ts [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+  Request.get("https://sagesse.polytech.umontpellier.fr/syllabus/"+subjectId+"/"+year, (error: any, response: any, body: any) => {
```
Comment (MartinCayuelas) [inline src/controllers/sagesseController.ts]: same here for res

File: src/routes/router.ts [hunk 3]
```diff
@@ -5,11 +5,13 @@ import homeRoute from './home';
 import igRoute from './ig';
 import doRoute from './do';
 import contactRoute from './contact';
+import sagesseRouter from './sagesse';
```
Comment (MartinCayuelas) [inline src/routes/router.ts]: should be sagesseRoute :+1:

File: src/routes/sagesse.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
+import { getSubjectDetails, getModuleDetails, getSemesterDetails, getYearDetails, getFormationDetails } from '../controllers/sagesseController';
+
+import { Router } from 'express';
+const sagesseRouter = Router();
+
+
+//send informations about a formation and the id of its year
+sagesseRouter.get('/:formation/', (req: any, res: any) => {
```
Comment (MartinCayuelas) [inline src/routes/sagesse.ts]: req and res not any but Request and Response type like i said above :)

File: src/controllers/sagesseController.ts [hunk 5]
```diff
@@ -0,0 +1,234 @@
+const Request = require('request')
+const jsdom = require('jsdom')
```
Comment (MartinCayuelas) [inline src/controllers/sagesseController.ts]: I forgot to say that you can use other form of import! Convert it to ES6 :)

File: src/controllers/sagesseController.ts [hunk 6]
```diff
@@ -1,234 +1,270 @@
...
-  })
+    res.send(details); //Send the response
+  });
 };
 
 
-export const getYearDetails = async (yearId: string, year: string, res: any) => {
-  Request.get("https://sagesse.polytech.umontpellier.fr/syllabus/"+yearId+"/"+year, (error: any, response: any, body: any) => {
+export const getYearDetails = async (yearId: string, year: string, res: Response) => {
+  Request.get('https://sagesse.polytech.umontpellier.fr/syllabus/'+yearId+'/'+year, (error: any, response: ResponseReq, body: any) => {
```
Comment (MartinCayuelas) [inline src/controllers/sagesseController.ts]: error and body shoul have a type also

File: src/controllers/sagesseController.ts [hunk 7]
```diff
@@ -1,234 +1,270 @@
...
-      "TP":TP ,
-      "Projet":projet ,
-      "Terrain":terrain ,
-      "description":description ,
-      "context":context ,
-      "content":content ,
-      "semesters":semesters }
+    const dom = new JSDOM(body);
+
+    let details : any = {};
```
Comment (MartinCayuelas) [inline src/controllers/sagesseController.ts]: here also

(In most of cases we should use a real type and not any ) ;)

File: src/controllers/sagesseController.ts (hunk 6 above)
Comment (raphell) [reply src/controllers/sagesseController.ts]: I add the type "string" for body, and the type "object" for error

File: src/controllers/sagesseController.ts (hunk 7 above)
Comment (raphell) [reply src/controllers/sagesseController.ts]: I add the "object" type for details

**Comment by MartinCayuelas** [comment]
For the database it is necessary to go through the access codes in fact.

I didn't see but It's not good.

**Comment by MartinCayuelas** [comment]
Don't forget to run " npm run lint"

File: src/controllers/sagesseController.ts [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+export const getFormationDetails = async (nomFormation : string) => {
+  console.log("BEGIN of get formation")
+
+  let formationInfo = await db.query('SELECT * \
+                                       FROM sagesse.parcours p\
+                                       WHERE p."codParcours" = :codP ; '
+                                      , { replacements: {codP:nomFormation},
+                                          type: QueryTypes.SELECT});
+    let formationDetails = {
+      "id": formationInfo[0].idParcours ,
```
Comment (MartinCayuelas) [inline src/controllers/sagesseController.ts]: single quote

**Pull request merged**